### PR TITLE
enable the user to load the local database to avoid unnecessary data preprocessing 

### PR DESCRIPTION
**Pull request by XuhuiRen**
## Type of Change

add a parm "override" to the retrieval agent.

Previously, the user always needed to wait for a long time to create a local database. 

This parm could enable the user to avoid this data processing period and directly use the local database, which could potentially save a lot of time.

### CODE REVIEW

**Review by hshen14 — APPROVED** [review]

**Comment by hshen14** [comment]
Can you enable UT to improve the code coverage?

**Comment by XuhuiRen** [comment]
> Can you enable UT to improve the code coverage?

UT for this parameter "override" is added

**Comment by airMeng** [comment]
please use rebase not merge

**Comment by XuhuiRen** [comment]
> please use rebase not merge

 error operation, fixed

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/indexing/indexing.py [hunk 1]
```diff
@@ -137,4 +139,16 @@ def KB_construct(self, input):
...
+    def load(self, input):
+        if self.retrieval_type == "dense":
+            embedding = HuggingFaceInstructEmbeddings(model_name=self.embedding_model)
+            vectordb = Chroma(persist_directory=self.persist_dir, embedding_function=embedding)
+        else:
+            if self.document_store == "inmemory":
+                vectordb = self.KB_construct(input)
+            else:
+                vectordb = ElasticsearchDocumentStore(host="localhost", index=self.index_name,
+                                                            port=9200, search_fields=["content", "title"])
```
Comment (a32543254) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/indexing/indexing.py:152]: port 9200 is default for elasticsearch api, but what if port 9200 is occupy by other app?

Comment (XuhuiRen) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/indexing/indexing.py:152]: ok, i will consider to add parms on "host" and "port"

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py [hunk 2]
```diff
@@ -32,7 +32,6 @@ def __init__(self, persist_dir="./output", process=True, input_path=None,
         self.retrieval_type = retrieval_type
         self.retriever = None
         self.intent_detector = IntentDetector()
-        assert os.path.exists(input_path)==True, "You should give a specific path for the uploaded files!"
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py]: why removing the assertion?

**Review by a32543254 — APPROVED** [review]
LGTM

**Comment by XuhuiRen** [comment]
test donot merge

**Comment by XuhuiRen** [comment]
Using https://github.com/intel/intel-extension-for-transformers/pull/358 to solve the issue of UT test location error.

**Pull request closed without merging**